### ISSUE TITLE
Client has no Internet access after connected to Repeater

### ISSUE THREAD

**Issue by watoan** [Mygod/VPNHotspot#65]
Hello, I am new to this and using VPN Hotspot on OnePlus 6T(Device A) with Android Pie.

I tried to share my VPN to my another mobile(Device B) by WiFi-Hotspot:
1. Install VPN Hotspot on Device A
2. Add the VPN connection on Device and connected
3. Check the public IP by going myip.com and confirmed traffic from Device A going out via VPN route.
4. Turn on the repeater in VPN Hotspot
5. On Device B, connect to the Repeater network SSID of Device A
6. Device B able to get the IP 192.168.49.151 with gateway 192.168.49.1, but showing "No Internet Access" and unable to ping to the public

Are there something I missed? I would like to route all traffic from Device B via Device A VPN connection.

Thanks.

**Comment by Mygod** [comment]
Export debug information while B is connected to A.

**Comment by watoan** [comment]
[vpnhotspot-3972168543432983997.log](https://github.com/Mygod/VPNHotspot/files/2785618/vpnhotspot-3972168543432983997.log)

**Comment by Mygod** [comment]
Try using clean/reapply routings in the settings first.

If that doesn't fix it, try this command with adb and see if that fixes it for you.

```sh
adb shell su -c 'iptables -A vpnhotspot_fwd -o p2p0 -j ACCEPT'
```

**Comment by watoan** [comment]
It works perfectly after press the "Clean / reapply routing rules"

Many thanks!

**Comment by charliesz** [comment]
Hello,
I got the same problem. The client has no internet connection while I'm using a vpn (running the build-in vpn manager on my oneplus 5t)
I tried the "clean/reapply"-button several times with no success.
With a terminal emulator I exicuted this command:
`su -c 'iptables -A vpnhotspot_fwd -o p2p0 -j ACCEPT'
`

No success.

Here is my log file: [vpnhotspot-3236555473133012140.log](https://github.com/Mygod/VPNHotspot/files/2921879/vpnhotspot-3236555473133012140.log)

Thanks in advance
